language/libra-vm/src/libra_transaction_validator.rs

```diff
@@ -113,73 +70,38 @@ impl VMValidator for LibraVMValidator {
         transaction: SignedTransaction,
         state_view: &dyn StateView,
     ) -> VMValidatorResult {
-        let data_cache = StateViewCache::new(state_view);
         let _timer = TXN_VALIDATION_SECONDS.start_timer();
-        let gas_price = transaction.gas_unit_price();
-        let currency_code =
-            match account_config::from_currency_code_string(transaction.gas_currency_code()) {
-                Ok(code) => code,
-                Err(_) => {
-                    return VMValidatorResult::new(
-                        Some(StatusCode::INVALID_GAS_SPECIFIER),
-                        gas_price,
-                        GovernanceRole::NonGovernanceRole,
-                    )
-                }
-            };
-
         let txn_sender = transaction.sender();
-        let signature_verified_txn = if let Ok(t) = transaction.check_signature() {
+
+        let txn = if let Ok(t) = transaction.check_signature() {
             t
         } else {
             return VMValidatorResult::new(
                 Some(StatusCode::INVALID_SIGNATURE),
-                gas_price,
+                0,
```

Comment:
we will have a mempool behavior change here, but I think this one makes sense

Reply:
Does it actually change something? I was assuming that an invalid transaction will be discarded so that the priority for mempool will be irrelevant.